test/powershell/Language/Scripting/DeserializedTypeConversion.Tests.ps1

```diff
@@ -0,0 +1,271 @@
+        {
+            get { return name; }
+            set { name = value; }
+        }
+
+        public int Port
+        {
+            get { return port; }
+            set { port = value; }
+        }
+
+        public string ScriptText
+        {
+            get { return scriptText; }
+            set { scriptText = value; }
+        }
+    }
+'@
+
+    $type3 = Add-Type -PassThru -TypeDefinition @'
+    public class test3
+    {
+        private string name = "default";
+        private int port = 80;
+        private string scriptText = "1..6";
+
+        public string Name
+        {
+            get { return name; }
+            set { name = value; }
+        }
+
+        public int Port
+        {
+            get { return port; }
+            set { port = value; }
+        }
+
+        public string ScriptText
+        {
+            get { return scriptText; }
+        }
+    }
+'@
+
+$type4 = Add-Type -PassThru -TypeDefinition @'
+public class test4
+{
+    private string name = "default";
+    private int port = 80;
+    private string scriptText = "1..6";
+
+    public string Name
+    {
+        get { return name; }
+        set { name = value; }
+    }
+
+    public int Port
+    {
+        get { return port; }
+        set { port = value; }
+    }
+
+    internal void Compute()
+    {
+        scriptText = scriptText + " Computed";
+    }
+}
+'@
+
+    function test-1
+    {
+        param(
+            [parameter(position=0, mandatory=1)]
+            [test1] $test
+        )
+
+        $test | Format-List | Out-String
+    }    
+
+    function test-2
+    {
+        param(
+            [parameter(position=0, mandatory=1)]
+            [test2] $test
+        )
+
+        $test | Format-List | Out-String
+    }
+
+    function test-3
+    {
+        param(
+            [parameter(position=0, mandatory=1)]
+            [test3] $test
+        )
+
+        $test | Format-List | Out-String
+    }
+
+    function test-4
+    {
+        param(
+            [parameter(position=0, mandatory=1)]
+            [test4] $test
+        )
+
+        $test | Format-List | Out-String
+    }
+
+    $t1 = new-object test1 -Property @{name="TestName1";port=80;scriptText="1..5"}
+    $t2 = new-object test2 -Property @{Name="TestName2";Port=80;ScriptText="1..5"}
+    $t3 = new-object test3 -Property @{Name="TestName3";Port=80}
+    $t4 = new-object test4 -Property @{Name="TestName4";Port=80}
+    }
+    
+
+    AfterAll {        
+        Remove-Item $file -force -ea silentlycontinue
+        # Clean up.
+        Get-PSSession | Remove-PSSession
+    } 
+
+    It 'T1' { 
+
+        Export-Clixml -InputObject $t1 -Path $file
+        $dst1 = Import-Clixml $file
+
+        # Type casts should *succeed*.
+        { $tc1 = [test1]$dst1 }| Should not Throw
+
+        # Parameter bindings should *succeed*.
+        { test-1 $dst1 } | Should Not Throw
+    }
+
+    It 'T1 Test remoting scenario.'  -skip:$IsCoreCLR {
+        $s = New-PSSession
+
```

Comment:
Are we adding useful coverage here? Remoting can be slow and isn't always reliable. I'd like to keep remoting out of tests where possible.